src/blockdigest.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "blockdigest.h"
+#include "streams.h"
+#include "version.h"
+#include "uint256.h"
+#include "primitives/transaction.h"
+#include "utiltime.h"
+#include "utilstrencodings.h"
+
+namespace DigestFilter {
+
+std::vector<uint8_t> base_filter::outpoint_data(const COutPoint& outpoint) const
+{
+    CDataStream stream(SER_NETWORK, PROTOCOL_VERSION);
+    stream << outpoint;
+    return std::vector<uint8_t>(stream.begin(), stream.end());
+}
+
+std::vector<const std::vector<uint8_t>> base_filter::tx_data_vec(const CTransaction& tx) const
+{
+    std::vector<const std::vector<uint8_t>> tx_data;
```

Comment:
In theory, it's supposed to be the exact same stuff that goes into the bloom filters used up to this point. In practice, I have no idea what's going on with these 1102's and 01's and stuff. I suspect the original bloom filters also have these but I could be wrong. Should probably look into that.